packages/template-retail-react-app/app/contexts/index.js

```diff
@@ -27,23 +29,162 @@ import PropTypes from 'prop-types'
  *
  * import {useCategories} from './hooks'
  *
- * const {categories, setCategories} = useCategories()
+ * const {root, findAndModifyFirst, itemsKey, fetchCategoryNode} = useCategories()
  *
  */
 export const CategoriesContext = React.createContext()
-export const CategoriesProvider = ({categories: initialCategories = {}, children}) => {
-    const [categories, setCategories] = useState(initialCategories)
+export const CategoriesProvider = ({treeRoot = {}, children, locale}) => {
```

Comment:
Have we thought about what it might look like for the hook integration? This component seems to grown from a one liner to a pretty complicated one. I'm worried that we might end up re-write the logic during the hook integration, what can we do to mitigate that risk? (Sorry, I wish i could have said this at the beginning of the work, not when this is almost done..)

Reply:
The change should be all located in one spot, and if we can't call a hook inside a `useEffect` then I think we've uncovered a problem with the hooks library in that we still need the SDK to make calls like this. I expect we might need to continue to use the SDK in some places. If that's true, then this might be one of those spots where we don't use the hooks library